app/frontend/src/features/messages/useOnVisibleEffect.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from "react";
+import { useInView } from "react-intersection-observer";
+
+export default function useOnVisibleEffect<Param>(
+  param: Param,
+  onVisible?: (param: Param) => void
+) {
```

Comment:
```suggestion
export default function useOnVisibleEffect(
  onVisible?: () => void
) {
```
Now I'm seeing this being used the generic way, I think this might make more sense instead, since the components that use this can pass the info they pass through via `param` right now directly with a wrapper function like `() => onVisible(message.messageId)` etc